### DISCUSSION TITLE
AutoConnect to offer captive portal for 5min even if ESP32 is WiFi-Connected

### DISCUSSION THREAD

**Discussion by MudassirTufail** [Hieromon/AutoConnect#605] [How can I?]
Firstly, my thanks in advance for picking my question:

After reading (and attempting) https://hieromon.github.io/AutoConnect/adcpcontrol.html for days, I am still at loss to achieve the following:

1) AutoConnect to keep running captive portal if ESP32 is not WiFi-connected (this piece is good, and is a default behavior)
And
2) AutoConnect to keep running captive portal for first 5 min after ESP32 starts and WiFi-connected. That means ESP32 stays on SoftAP mode and user can still access the captive portal at 192.168.4.1 while ESP32 is already connected to a WiFi. I am looking to implement this behavior for first 5 min after ESP32 starts. This provides a time window (5min) if users wants to connect ESP32 to a different WiFi.

I do see options such as retain, timeout etc. but unfortunately I am not smart enough to use them correctly in my sketch. If someone can help me out with some sketch examples to achieve the above (especially "2"). Thank You

**Comment by Hieromon** [comment]
Captive portal activation is not possible after the 1st-WiF::ibegin connection successful. Therefore, your request "the captive portal will be activated if WiFi is unconnected, but the captive portal will be activated even if there is an existing WiFi connection" cannot be realized.

**Reply by supiiik** [reply]
I'm dealing with similar problem. Is there a way to let AP mode activated, when AutoConnect connect to AP? Something like mode("WIFI_AP_STA") so I can connect to ESP32 with my phone even if ESP is already connected to AP?

**Reply by linaspasv** [reply]
Lets say we have a successful connection to the WiFi. Is it possible to somehow access the portal so we could re-configure (disconnect / change) an existing WiFi connection?
